### ISSUE TITLE
Live reload is broken starting with Spring Boot 2.5.2

### ISSUE THREAD

**Issue by pleku** [vaadin/flow#11375]
Using e.g. https://github.com/vaadin/skeleton-starter-flow-spring
Bump the Spring Boot version to 2.5.2 in pom.xml
when running the application in any way
and then opening it in the browser
The websocket connection for live reload fails on application startup with the line in
```ts
this.webSocket = new WebSocket(url);
```
https://github.com/vaadin/flow/blob/master/flow-client/src/main/frontend/VaadinDevmodeGizmo.ts#L1014 

with the exception
```javascript
msg: Event
bubbles: false
cancelBubble: false
cancelable: false
composed: false
currentTarget: WebSocket
binaryType: "blob"
bufferedAmount: 0
extensions: ""
onclose: (_) => {…}
onerror: (err) => this.handleError(err)
onmessage: (msg) => this.handleMessage(msg)
onopen: null
protocol: ""
readyState: 3
url: "ws://localhost:35729/"
__proto__: WebSocket
defaultPrevented: false
eventPhase: 2
isTrusted: true
path: []
returnValue: true
srcElement: WebSocket
binaryType: "blob"
bufferedAmount: 0
extensions: ""
onclose: (_) => {…}
onerror: (err) => this.handleError(err)
onmessage: (msg) => this.handleMessage(msg)
onopen: null
protocol: ""
readyState: 3
url: "ws://localhost:35729/"
__proto__: WebSocket
target: WebSocket {url: "ws://localhost:35729/", readyState: 3, bufferedAmount: 0, onopen: null, onerror: ƒ, …}
timeStamp: 10127.399999976158
type: "error"
__proto__: Event
```

Downgrading to Spring Boot 2.5.1 helps.

The only change in 2.5.2 that looks related like a probably cause is https://github.com/spring-projects/spring-boot/commit/b75c79cc7c2701e4d7c25349b458f84b3cf88648

**Comment by denis-anisimov** [comment]
Cannot reproduce.

Live reload works. There is gizmo, no server side exceptions, no client side exceptions.

I had this project already but when after pull the whole project build failed.
So I jut removed it completely and cloned it from scratch.

Steps to reproduce (exact step by step):
- git clone git@github.com:vaadin/spring.git
- mvn clean install -DskipTests
- mvn spring-boot:run
- open browser page http://localhost:8080/

The result is:
 - no any exception in the server log
 - no any error in the browser console log
 - dev mode Gizmo is rolled up and I can open it and see the info inside it

**Comment by pleku** [comment]
@denis-anisimov the bump of spring boot version was missing from the list, it is now included.

**Comment by denis-anisimov** [comment]
I don't see what can be done on our side.
This looks like bug in Spring: https://github.com/spring-projects/spring-boot/issues/27147

JS WebSocket class can't connect to the server and there is nothing to fix on our side.
The same happens with:
- console code ws = new WebSocket("ws://localhost: 35729");
- Live reload Chrome extension: it doesn't work with 2.5.2

**Comment by pleku** [comment]
Fixed in Spring Boot 2.4.9 & 2.5.3
